engine/src/main/scala/cromwell/io/ServiceRouterActor.scala

```diff
@@ -0,0 +1,48 @@
+package cromwell.io
+
+import java.io.IOException
+
+import akka.actor.SupervisorStrategy.{Escalate, Restart}
+import akka.actor.{Actor, ActorInitializationException, OneForOneStrategy, Props}
+import akka.routing.SmallestMailboxPool
+import com.typesafe.config.Config
+import cromwell.core.Dispatcher
+import cromwell.services.io.IoActorCommand
+import net.ceedubs.ficus.Ficus._
+
+/**
+  * Wrapper for the IO router actor.
+  * This is a sub optimal level of indirection as messages could directly be directed to the router
+  * instead of having to go through this actor.
+  * The reason is that the Io Actor is under the umbrella of the service registry which instantiates its services generically using reflection,
+  * which can't work for a router actor.
+  * Was the router IoActor to become a first class citizen, this wrapper should be removed.
```

Comment:
I don't picture this as being part of the service registry - those things should be stuff that one would be reasonably expected to drop in some replacement (e.g. a different DB for metadata or KV). 

Having it top level does bring up the issue of how to have both the backends and engine talking to this thing however.